src/store/blocks.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable no-param-reassign */
+/*
+   This file is part of Astarte.
+
+   Copyright 2021 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import AstarteClient, { AstarteBlock, AstarteCustomBlock } from 'astarte-client';
+import _ from 'lodash';
+
+import type { StoreState } from './index';
+
+interface BlocksState {
+  blocks: AstarteBlock[] | null;
+  status: 'empty' | 'loading' | 'ok' | 'err';
+  error: string | null;
+  errorByBlock: { [blockName: string]: string };
+  blocksBeingRegistered: string[];
+  blocksBeingUpdated: string[];
+  blocksBeingDeleted: string[];
+}
+
+const initialState: BlocksState = {
+  blocks: null,
+  status: 'empty',
+  error: null,
+  errorByBlock: {},
+  blocksBeingRegistered: [],
+  blocksBeingUpdated: [],
+  blocksBeingDeleted: [],
+};
+
+const asyncActions = {
+  getList: createAsyncThunk('blocks/getList', (params, { extra }) =>
+    (extra as AstarteClient).getBlocks(),
+  ),
+  get: createAsyncThunk('blocks/get', (name: string, { extra }) =>
```

Comment:
If this the **Blocks** state, should we add getters for single blocks too?
Not sure, but i think it's worth splitting the single block from the collection of blocks.